reviews/review-03+2020-11-30.md

```diff
@@ -0,0 +1,169 @@
+in a way that more closely resembles how an RSE might create a project. This
+might suggest for example starting with Git to get things organized and under
+version control right away. Does chapter 7 on working with teams belong earlier
+as well? It felt orienting to see the chapter on automating analyses just after
+chapter on teamwork. I am thinking about the behaviors that go along with
+software development and placing emphasis on those in a clear context. I'm not
+convinced the current order is the best possible one. Perhaps storyboarding the
+entire book would help the authors discover opportunities to reorganize. The
+approach I suggest is a stylistic preference and might mean making chapters less
+monolithic, returning to topics as they are needed.
+
+b. Sec 1.1: Mentions "three" related concepts; why not give sustainability its
+own bullet point? I love your definition.
+
+c. Sec 1.1: Grammatical rewrite needed: "Nobody argues that research should be
+irreproducible or unsustainable, but "not against it" and actively supporting it
+are very different things. Academia doesn't yet know reward people for writing
+useful software, so while you may be thanked, the effort you put in may not
+translate into job security or decent pay."
+
+d. Sec 1.2: I assume the audience are learner personas---perhaps ones that will
+be used throughout the book. This needs explanation because it is jarring to
+just have three "audience (members? personas?)" just pop into existence on the
+page. Why not explain the use of the personas and how it benefits the reader to
+relate to them at this point in the text?
+
+e. Sec 1.2.1: With the exception of your clearly stating Python 3 is needed,
+might it be better to specify versions that were used in the creation of the
+book examples? I think it is fine to allow learners freedom, especially since at
+this point, they would not be novices and probably have their own preferences. I
+might still want these specifics (especially if I were teaching this material)
+and those could be specified in detail in the GitHub repo accompanying the
+book. I don't seem to find Appendix E in my PDF version of the manuscript.
+
+f. Sec 1.3: I might add a sentence or two generalizing the Zipf's law
+example. If you want to "sell" the book, I think I would emphasize how looking
+for patterns in data is common to most scientific analyses and so you have
+chosen a straightforward example that will allow you to teach the concrete
+objectives you state. Rather than Syllabus, maybe something more direct---"What
+you will learn."
+
+g. Given what I've read up the Chapter 2 on the Unix shell, I am under the
+assumption that the audience of this book is already using Python at least an
+intermediate level. I'd like to believe that this level of user at least knows
+how to manipulate Linux directories. Everything in this chapter felt like
+prerequisite, so perhaps it should be an appendix?
+
```

Comment:
I tend to agree that for most people in our target audience, the Basics of the Unix Shell chapter is probably mostly stuff they already know. Having said that:
* In my experience many people have trouble with relative vs absolute paths
* It can be nice to start with content that you're mostly familiar with to build confidence before moving into harder topics

I'm therefore probably happy to leave the Basics of the Unix Shell chapter as is, unless others have a strong opinion about moving it to an appendix.

Reply:
It's totally feasible to be an intermediate R programmer without ever having to really interact with the shell, is this different for Python users?  

We don't really introduce anything critical to the project in this chapter, except maybe an example of using `wc`, so it could be moved.  But, like @DamienIrving, I don't have strong feelings about this.

Reply:
From my grad school experience, there seems to exist many researchers who know there way around Python or R, but tend to avoid  the shell as much as possible. I don't think there is any harm in asking more advanced shell users to skip the shell basics chapter, but I would also be onboard with putting in the appendix and ask the novice shell users to read it before continuing in the book.

Reply:
I am a strong :-1: on major surgery at this point. No matter what we ship, someone will have been better served by some better selection or arrangement of topics. What we have now is coherent and useful - if it turns out we were badly wrong, we can ship Version 2 after the R book comes out.